.banzaicloud/charts/spotguide-modern-go-application/templates/deployment.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "spotguide-modern-go-application.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "spotguide-modern-go-application.name" . }}
+    helm.sh/chart: {{ include "spotguide-modern-go-application.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+{{- if .Values.autoscaling.enabled }}
+  replicas: {{ .Values.autoscaling.minReplicas }}
+{{- else }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "spotguide-modern-go-application.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "spotguide-modern-go-application.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+    spec:
+      volumes:
+        - name: config
+          configMap:
+            name: {{ include "spotguide-modern-go-application.fullname" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: app
+              containerPort: 8000
+              protocol: TCP
+            - name: instrumentation
+              containerPort: 10000
+              protocol: TCP
+          volumeMounts:
+            - name: config
+              mountPath: /config.yaml
+              subPath: config.yaml
+          livenessProbe:
+            httpGet:
+              path: /version
+              port: instrumentation
+          readinessProbe:
+            httpGet:
+              # path: /healthz
```

Comment:
WIP?

Reply:
This will work once MySQL is added in a subsequent PR.